lldb/test/API/functionalities/data-formatter/type-recognizer/TestTypeRecognizerDelete.py

```diff
@@ -0,0 +1,71 @@
+"""
+Test lldb data formatter subsystem.
+"""
+
+
+from lldbsuite.test.lldbtest import *
+
+
+class TypeRecognizerListTestCase(TestBase):
+    def test_type_recognizer_delete(self):
+        """Test that the 'type recognizer delete' command works properly"""
+
+        self.runCmd("type recognizer add Foo -F testDefault")
+        self.runCmd("type recognizer add Foo -F testFooCat -w FooCat")
+        self.runCmd("type recognizer add Bar -F testDefault")
+        self.runCmd("type recognizer add Bar -F testBarCat -w BarCat")
+        self.runCmd("type recognizer add Bar -F testBarCatSecond -w BarCatSecond")
+
+        self.runCmd("type recognizer delete Foo")
+        self.expect(
+            "type recognizer list",
+            substrs=[
+                "Bar:  Python function testDefault",
+                "Bar:  Python function testBarCat",
+                "Bar:  Python function testBarCatSecond",
+                "Foo:  Python function testFooCat",
```

Comment:
It seems that even when the command doesn't specify category, the default one is implicitly specified. So the `delete` command in this test doesn't touch recognizers for `Foo` in categories other that the default one. Is that correct?

Reply:
Yes, it's correct.